### PR TITLE
Move common configurations to TuningConfig

### PR DESCRIPTION
**Pull request by liran-funaro**
### Description

The goal of this PR is to reduce code duplications and redundant code.
It does not affect Druid's UX. Only the internal code structure.

* Move common methods that are used in `HadoopTuningConfig` and in `AppenderatorConfig` to `TuningConfig`
* Rename `rowFlushBoundary` in `HadoopTuningConfig` to `maxRowsInMemory` to match `TuningConfig` API
  - This was already defined as `@JsonProperty("maxRowsInMemory")`, but the method name did not match `TuningConfig` API

<hr>

This PR has:
- [X] been self-reviewed.
- [X] added documentation for new or modified features or behaviors.
- [X] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/licenses.yaml)
- [X] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [X] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [X] been tested in a test Druid cluster.

<hr>

##### Key changed/added classes in this PR
 * `TuningConfig`
 * `AppenderatorConfig`
 * `HadoopTuningConfig`

### CODE REVIEW

File: extensions-contrib/materialized-view-maintenance/src/test/java/org/apache/druid/indexing/materializedview/MaterializedViewSupervisorTest.java [hunk 1]
```diff
@@ -317,6 +318,32 @@ public void testCheckSegmentsAndSubmitTasks() throws IOException
 
   }
 
+  @Test
+  public void testCreateTask()
```
Comment (Eshcar) [inline extensions-contrib/materialized-view-maintenance/src/test/java/org/apache/druid/indexing/materializedview/MaterializedViewSupervisorTest.java:322]: shortly explain what this is testing in the PR

**Review by Eshcar — APPROVED** [review]
LGTM
All changes seem to improve code quality
can you just add a sentence (in the jira) explaining this is a preamble to a later PR that will add the I2 extension point

**Review by clintropolis — APPROVED** [review]

**Comment by clintropolis** [comment]
I don't seem to be able to retry the failing CI which looks like it was a flaky test.

**Comment by liran-funaro** [comment]
Thanks, @clintropolis.
I appreciate your help merging this PR.
Please also check out the follow-up PRs: #10593 and/or #10001.

**Pull request merged**